master/internal/rm/kubernetesrm/job.go

```diff
@@ -0,0 +1,751 @@
+package kubernetesrm
```

Comment:
Reviewer note: this code started as a copy of the old `pod.go` (but modified so heavily github's diff algorithm switch from calling it a rename+changes to a delete+rewrite). `jobs.go` is the same, copied from `pods.go`. I'm calling this out to say: I refactored the code as I thought it was necessary and as it made me more confident in its correctness but I didn't go a lot further. I'll probably do a style-oriented refactor once this PR is in (it's all going to a feature branch for now).